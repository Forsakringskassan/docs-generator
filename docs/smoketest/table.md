---
title: "Smoketest: tables"
short-title: Tables
layout: article
---

## Narrow table

Narrow on all screen sizes.

| A   | B   | C   |
| --- | --- | --- |
| a   | b   | c   |
| d   | e   | f   |
| g   | h   | i   |

## Wide table

Wide on most screen sizes.

| A                          | B                          | C                          |
| -------------------------- | -------------------------- | -------------------------- |
| Lorem ipsum dolor sit amet | Lorem ipsum dolor sit amet | Lorem ipsum dolor sit amet |
| Lorem ipsum dolor sit amet | Lorem ipsum dolor sit amet | Lorem ipsum dolor sit amet |
| Lorem ipsum dolor sit amet | Lorem ipsum dolor sit amet | Lorem ipsum dolor sit amet |

## Ultra wide table

Wide on all screen sizes.

| A                                                        | B                                                        | C                                                        |
| -------------------------------------------------------- | -------------------------------------------------------- | -------------------------------------------------------- |
| Lorem ipsum dolor sit amet, consectetur adipiscing elit. | Lorem ipsum dolor sit amet, consectetur adipiscing elit. | Lorem ipsum dolor sit amet, consectetur adipiscing elit. |
| Lorem ipsum dolor sit amet, consectetur adipiscing elit. | Lorem ipsum dolor sit amet, consectetur adipiscing elit. | Lorem ipsum dolor sit amet, consectetur adipiscing elit. |
| Lorem ipsum dolor sit amet, consectetur adipiscing elit. | Lorem ipsum dolor sit amet, consectetur adipiscing elit. | Lorem ipsum dolor sit amet, consectetur adipiscing elit. |

## In details

::: details Narrow

| A   | B   | C   |
| --- | --- | --- |
| a   | b   | c   |
| d   | e   | f   |
| g   | h   | i   |

:::

::: details Wide

| A                          | B                          | C                          |
| -------------------------- | -------------------------- | -------------------------- |
| Lorem ipsum dolor sit amet | Lorem ipsum dolor sit amet | Lorem ipsum dolor sit amet |
| Lorem ipsum dolor sit amet | Lorem ipsum dolor sit amet | Lorem ipsum dolor sit amet |
| Lorem ipsum dolor sit amet | Lorem ipsum dolor sit amet | Lorem ipsum dolor sit amet |

:::

::: details Ultra wide

| A                                                        | B                                                        | C                                                        |
| -------------------------------------------------------- | -------------------------------------------------------- | -------------------------------------------------------- |
| Lorem ipsum dolor sit amet, consectetur adipiscing elit. | Lorem ipsum dolor sit amet, consectetur adipiscing elit. | Lorem ipsum dolor sit amet, consectetur adipiscing elit. |
| Lorem ipsum dolor sit amet, consectetur adipiscing elit. | Lorem ipsum dolor sit amet, consectetur adipiscing elit. | Lorem ipsum dolor sit amet, consectetur adipiscing elit. |
| Lorem ipsum dolor sit amet, consectetur adipiscing elit. | Lorem ipsum dolor sit amet, consectetur adipiscing elit. | Lorem ipsum dolor sit amet, consectetur adipiscing elit. |

:::
