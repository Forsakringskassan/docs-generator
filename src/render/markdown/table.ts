/* eslint-disable camelcase -- upstream library uses snake_case */

import type markdownIt from "markdown-it";

export function table(): (md: markdownIt) => void {
    return function (md: markdownIt): void {
        md.renderer.rules.table_open = () => `<table class="table">\n`;
        md.renderer.rules.tr_open = () => `<tr class="table__row">\n`;
        md.renderer.rules.th_open = () =>
            `<th class="table__column table__column--text" scope="col">`;
        md.renderer.rules.td_open = () =>
            `<td class="table__column table__column--text">`;
    };
}
