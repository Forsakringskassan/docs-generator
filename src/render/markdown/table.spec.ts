import markdownIt from "markdown-it";
import { table } from "./table";

const md = markdownIt();
md.use(table());

it("should render table with datatable classes", () => {
    expect.assertions(1);
    const markup = md.render(`
| name           | age |
|----------------|-----|
| Donald Duck    | 47  |
| Scrooge McDuck | 81  |
`);
    expect(markup).toMatchSnapshot();
});
