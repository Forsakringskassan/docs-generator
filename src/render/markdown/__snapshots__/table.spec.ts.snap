// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render table with datatable classes 1`] = `
"<table class="table">
<thead>
<tr class="table__row">
<th class="table__column table__column--text" scope="col">name</th>
<th class="table__column table__column--text" scope="col">age</th>
</tr>
</thead>
<tbody>
<tr class="table__row">
<td class="table__column table__column--text">Donald Duck</td>
<td class="table__column table__column--text">47</td>
</tr>
<tr class="table__row">
<td class="table__column table__column--text">Scrooge McDuck</td>
<td class="table__column table__column--text">81</td>
</tr>
</tbody>
</table>
"
`;
